pkg/compactor/block_upload.go

```diff
@@ -262,29 +285,73 @@ func (c *MultitenantCompactor) UploadBlockFile(w http.ResponseWriter, r *http.Re
 	w.WriteHeader(http.StatusOK)
 }
 
-func (c *MultitenantCompactor) completeBlockUpload(ctx context.Context, logger log.Logger, userBkt objstore.Bucket, blockID ulid.ULID, meta metadata.Meta) error {
+func (c *MultitenantCompactor) validateAndCompleteBlockUpload(logger log.Logger, userBkt objstore.Bucket, blockID ulid.ULID, meta *metadata.Meta) {
 	level.Debug(logger).Log("msg", "completing block upload", "files", len(meta.Thanos.Files))
 
-	// Upload meta file so block is considered complete
+	ch := make(chan struct{})
+	var wg sync.WaitGroup
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// start a go routine that updates the validation file's timestamp every heartbeat interval
+	wg.Add(1)
+	go c.periodicValidationUpdater(ctx, logger, blockID, userBkt, ch, &wg, cancel)
```

Comment:
nit: we could reduce the scope for `wg` by calling `Done` here, instead of passing to `periodicValidationUpdater`.

```suggestion
	go func(){ defer wg.Done(); c.periodicValidationUpdater(ctx, logger, blockID, userBkt, ch, cancel) }()
```

Reply:
Nice, changed it.